src/couch_replicator.erl

```diff
@@ -129,3 +135,53 @@ cancel_replication(RepId, #user_ctx{name = Name, roles = Roles}) ->
             {error, not_found}
         end
      end.
+
+
+-spec stream_terminal_docs_info(binary(), user_doc_cb(), any()) -> any().
+stream_terminal_docs_info(Db, Cb, UserAcc) ->
+    DDoc = <<"_replicator">>,
+    View = <<"terminal_states">>,
+    QueryCb = fun handle_replicator_doc_query/2,
+    Args = #mrargs{view_type = map, reduce = false},
+    Acc = {Db, Cb, UserAcc},
+    try fabric:query_view(Db, DDoc, View, QueryCb, Acc, Args) of
+    {ok, {Db, Cb, UserAcc1}} ->
+        UserAcc1;
+    {not_found, missing} ->
+        UserAcc
```

Comment:
I think this function should return `{ok, any()} | {error, db_missing | view_missing}` and be handled by the caller.